src/layout/TagsInput.js

```diff
@@ -7,6 +7,11 @@ import { observer, inject } from 'mobx-react'
 import { Flex } from './Flex'
 import Popover from './Popover'
 
+let isValidTag = (tag, tags) => {
+  let cleanTag = _.toLower(_.trim(tag))
+  return !_.isEmpty(cleanTag) && !_.some(t => _.toLower(t) === cleanTag, tags)
```

Comment:
We should not assume that all TagsInputs are case insensitive - the simplest thing I can think of is to expose a validateTag function like this as a prop and pass it in on TagsQuery. I suspect we can assume case insensitivity is fine for TagsQuery, but even then that isn't true for exact matches where case _does_ matter. Ultimately, I'm not sure it's even worth it to do this at all because of that.

Reply:
Ok removed the case check.

Reply:
A tag being valid is not the only thing `isValidTag` is checking, so the name is a bit off.

A more flexible approach would be to have the option to control this component with an `addTag: (tags, tag) => tags` prop, defaulting to what we've got here.

Also I think trim/lowercase is as good a default as any.

Reply:
@stellarhoof `addTag` is already a prop since this control doesn't have its own native state